core/domain/classifier_domain_test.py

```diff
@@ -76,6 +76,246 @@ def test_to_dict(self):
+                }
+            ],
+            'classifier_data': {},
+            'data_schema_version': 1
+        }
+        training_job_obj = self._get_training_job_from_dict(
+            training_job_dict)
+        training_job_dict['status'] = 'PENDING'
+        training_job_obj.update_status(training_job_dict['status'])
+        data_dict = training_job_obj.to_dict()
+        self.assertTrue(data_dict['status'])
+
+
+    def test_update_status_from_pendingtocomplete_with_successful_update(self):
+        training_job_dict = {
+            'job_id': 'exp_id1.SOME_RANDOM_STRING',
+            'algorithm_id': 'TextClassifier',
+            'interaction_id': 'TextInput',
+            'exp_id': 'exp_id1',
+            'exp_version': 1,
+            'next_scheduled_check_time':
+                datetime.datetime.strptime(
+                    '2017-08-11 12:42:31', '%Y-%m-%d %H:%M:%S'),
+            'state_name': 'a state name',
+            'status': 'PENDING',
+            'training_data': [
+                {
+                    'answer_group_index': 1,
+                    'answers': ['a1', 'a2']
+                },
+                {
+                    'answer_group_index': 2,
+                    'answers': ['a2', 'a3']
+                }
+            ],
+            'classifier_data': {},
+            'data_schema_version': 1
+        }
+        training_job_obj = self._get_training_job_from_dict(
+            training_job_dict)
+        training_job_dict['status'] = 'COMPLETE'
+        training_job_obj.update_status(training_job_dict['status'])
+        data_dict = training_job_obj.to_dict()
+        self.assertTrue(data_dict['status'])
+
+
+    def test_update_status_from_pending_to_failed_with_successful_update(self):
+        training_job_dict = {
+            'job_id': 'exp_id1.SOME_RANDOM_STRING',
+            'algorithm_id': 'TextClassifier',
+            'interaction_id': 'TextInput',
+            'exp_id': 'exp_id1',
+            'exp_version': 1,
+            'next_scheduled_check_time':
+                datetime.datetime.strptime(
+                    '2017-08-11 12:42:31', '%Y-%m-%d %H:%M:%S'),
+            'state_name': 'a state name',
+            'status': 'PENDING',
+            'training_data': [
+                {
+                    'answer_group_index': 1,
+                    'answers': ['a1', 'a2']
+                },
+                {
+                    'answer_group_index': 2,
+                    'answers': ['a2', 'a3']
+                }
+            ],
+            'classifier_data': {},
+            'data_schema_version': 1
+        }
+        training_job_obj = self._get_training_job_from_dict(
+            training_job_dict)
+        training_job_dict['status'] = 'FAILED'
+        training_job_obj.update_status(training_job_dict['status'])
+        data_dict = training_job_obj.to_dict()
+        self.assertTrue(data_dict['status'])
+
+
+    def test_update_status_with_exception(self):
```

Comment:
Rename to `test_update_status_raises_exception_if_status_change_is_invalid`.

Reply:
Done